pkg/resource/deploy/step_generator_test.go

```diff
@@ -360,3 +360,97 @@ func TestEngineDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceInheritsOptionsFromParent(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name              string
+		parentDeletedWith resource.URN
+		childDeletedWith  resource.URN
+		wantDeletedWith   resource.URN
+	}{
+		{
+			// Children missing DeletedWith should inherit DeletedWith
+			name:              "inherit",
+			parentDeletedWith: "parent-deleted-with",
+			childDeletedWith:  "",
+			wantDeletedWith:   "parent-deleted-with",
+		},
+		{
+			// Children with DeletedWith should not inherit DeletedWith
+			name:              "override",
+			parentDeletedWith: "parent-deleted-with",
+			childDeletedWith:  "this-value-is-set-and-should-not-change",
+			wantDeletedWith:   "this-value-is-set-and-should-not-change",
+		},
+		{
+			// Children with DeletedWith should not inherit empty DeletedWith.
+			name:              "keep",
+			parentDeletedWith: "",
+			childDeletedWith:  "this-value-is-set-and-should-not-change",
+			wantDeletedWith:   "this-value-is-set-and-should-not-change",
+		},
+	}
+
+	for _, tt := range tests {
+		test := tt
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			parentURN := resource.NewURN("a", "proj", "d:e:f", "a:b:c", "parent")
+			parentGoal := &resource.Goal{
+				Parent:      "",
+				Type:        parentURN.Type(),
+				Name:        parentURN.Name(),
+				DeletedWith: test.parentDeletedWith,
+			}
+
+			childURN := resource.NewURN("a", "proj", "d:e:f", "a:b:c", "child")
+			goal := &resource.Goal{
+				Parent:      parentURN,
+				Type:        childURN.Type(),
+				Name:        childURN.Name(),
+				DeletedWith: test.childDeletedWith,
+			}
+
+			sg := stepGenerator{
+				urns: map[resource.URN]bool{
+					parentURN: true,
+				},
+				resourceGoals: map[resource.URN]*resource.Goal{
+					parentURN: parentGoal,
+				},
+			}
+			err := sg.configureGoal(goal)
+
+			assert.Nil(t, err)
+			assert.Equal(t, goal.DeletedWith, test.wantDeletedWith)
+		})
+	}
+}
+
+// TestResourceInheritsOptionsFromParentNoGoal checks that configureGoal does not error when the parent goal is not
+// present in the step generator. This can occur for Read resources (i.e. pkg.Resource.get()) and MLCs.
+func TestResourceInheritsOptionsFromParentNoGoal(t *testing.T) {
+	t.Parallel()
+
+	parentURN := resource.NewURN("a", "proj", "d:e:f", "a:b:c", "parent")
+
+	childURN := resource.NewURN("a", "proj", "d:e:f", "a:b:c", "child")
+	goal := &resource.Goal{
+		Parent: parentURN,
+		Type:   childURN.Type(),
+		Name:   childURN.Name(),
+	}
+
+	// Create a step generator without the parent goal.
+	sg := stepGenerator{
+		urns: map[resource.URN]bool{
+			parentURN: true,
+		},
+		resourceGoals: map[resource.URN]*resource.Goal{},
+	}
+	err := sg.configureGoal(goal)
+
+	assert.Nil(t, err)
+}
```

Comment:
Catches the regression.

Reply:
Changed approach to track goals in the resource monitor.

Resources without goals should show up in the resource monitor (even MLCs). Read resources don't actually have resource options sent over the wire to propagate and expectation for imported resources to be inherited by child resources are an inaccuracy.